inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp

```diff
@@ -74,11 +76,39 @@ class StaticShapeNMS final : public StageNode {
         input5->serializeBuffer(serializer);
         outputData->serializeBuffer(serializer);
         outputDims->serializeBuffer(serializer);
+
+        if (tempBuffers().size())
+            tempBuffer(0)->serializeBuffer(serializer);
     }
 };
 
+bool isCMXEnough(int cmxSize, int numSlices, int spatDim, std::vector<int> bufferSizes) {
+    int curOffset = 0;
+    int curSlice = 0;
+
+    auto buffer_allocate = [&curOffset, &curSlice, &numSlices, cmxSize](int numBytes) {
+        if (curOffset + numBytes < cmxSize) {
+            curOffset += numBytes;
+        } else if (curSlice < numSlices && numBytes < cmxSize) {
+            curSlice++;
+            curOffset = numBytes;
+        } else {
+            return false;
+        }
+
+        return true;
+    };
+
+    for (auto s : bufferSizes) {
+        if (!buffer_allocate(s)) return false;
+    }
+
+    return true;
```

Comment:
Seems possibly to use `return std::all_of(bufferSizes.cbegin(), bufferSizes.cend(), buffer_allocate)`

Reply:
I guess it is `std::any_of` :)

Reply:
No, I think all right :) Please review it one more time)

Reply:
We can say, than we return true if `buffer_allocate` predicate always true, but return false if predicate false at least once.

Reply:
Actually both options are possible depending on what is functor: is it `buffer_allocates` or `!buffer_allocates`. At the same time, `std::any_of` reflects existing `for-loop` more directly.

Reply:
Fixed